src/rulesets/Base/arraymath.jl

```diff
@@ -104,3 +104,14 @@ function rrule(::typeof(\), b::Real, A::AbstractArray{<:Real})
     end
     return Y, backslash_pullback
 end
+
+#####
+##### Negation (Unary -)
+#####
+
+function rrule(::typeof(-), x::AbstractArray)
+    function negation_pullback(ȳ)
+        return NO_FIELDS, InplaceableThunk(@thunk(-ȳ), ā -> _subtract!!(ā, ȳ))
```

Comment:
If we made `add!!` promise only to call `InplaceThunk.add!` on mutable things then could do:
```suggestion
        return NO_FIELDS, InplaceableThunk(@thunk(-ȳ), ā -> ā .-= ȳ)
```